src/api.rs

```diff
@@ -4,13 +4,19 @@ use log::Event::*;
 use job::{Code, CodeImpl, Job};
 use std::sync::Arc;
 use thread_pool::{self, Registry, WorkerThread};
+use num_cpus;
 
 /// Initializes the Rayon threadpool. You don't normally need to do
 /// this, as it happens automatically, but it is handy for
 /// benchmarking purposes since it avoids initialization overhead in
 /// the actual operations.
-pub fn initialize() {
-    let registry = thread_pool::get_registry();
+/// But you need to call this if you want to specify the maximum number
+/// of threads that Rayon is allowed to use.
+/// If you don't specify it, Rayon uses all the cores avaiable (determined
```

Comment:
I think I would prefer if there was an easy way to get the default behavior; for example, this could take `Option<u32>`, or else we could say that if you say 0, we will autodetect the number of CPUs.